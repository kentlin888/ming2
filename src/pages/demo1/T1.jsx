import React, { Component } from 'react';
import T1_css from'./T1.css';

//var img_str = require("html-loader!./demo1.htm");
//let img_str = require("./demo1.htm")
import img_str from "./demo1.htm";
import old_street from './_old_street.jpg'
console.log("LOG: ~ file: T1.jsx ~ line 8 ~ old_street", old_street)
//console.log("LOG: ~ file: T1.jsx ~ line 5 ~ img_str", img_str)



function appendImg(){
    /*
    var img = document.createElement('img');
    img.src = img_url;
    img.className = "tune_svg_width";
    return img;
    */
    return img_str;
}
//let React = require('react')
class T1 extends React.Component {
    onLog111=() => {
        console.log(1111111)
    }
    render() {
        let imgPath = `../../assets/`
        return (
            <div>
                <div>PPP2229</div>
                <img src={`${imgPath}${old_street}`} alt=""/>
                <div>
                    <button id="btnLog111" data-testid="btnLog111" onClick={this.onLog111}>log 111111</button>
                </div>
                {/* <img src={old_street} alt=""/> */}
            </div>
        )
    }
}
//module.exports = <T1></T1>
export default <T1></T1>