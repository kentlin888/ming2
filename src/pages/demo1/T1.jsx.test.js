import React from 'react'
//import j1 from 'jest'
import Enzyme, { shallow, mount, render } from 'enzyme'
import toJson from 'enzyme-to-json';

import EnzymeAdapter from 'enzyme-adapter-react-16';
import T1 from './T1.jsx'
import a from './a'
import register from 'ignore-styles';
import { describe, expect, test } from '@jest/globals'
//let j2 = require('jest')

// import { expect } from 'chai';
register(['.css', '.sass', '.scss']);

const jsdom = require('jsdom');

const { JSDOM } = jsdom;
const { window } = new JSDOM('');
const { document } = (new JSDOM(``)).window;

/**@type {any} */
let globalAny = global;
globalAny.document = document;
globalAny.window = window;

require('@babel/polyfill')

Enzyme.configure({
    adapter: new EnzymeAdapter()
});

class X1 extends React.Component {
    render() {
        return (
            <div>
                AAA
            </div>
        )
    }
}
// /**
//  * Factory function to create a ShallowWrapper for App component.
//  * @function setup
//  * @param {object} props - Component props specific to this setup.
//  * @param {object} state - Initial state for setup.
//  * @returns {ShallowWrapper}
//  */
// const setup = ( props = {}, state = null) => {
//     const wrapper = shallow(<App {...props} />);
//     if (state) wrapper.setState(state);
//     return wrapper;
// };

describe('T1.jsx.test.js', () => {

    it('t1 demo.....2', () => {
        //let aa = <T1></T1>
        const component = mount(T1)
        let wrapper = shallow(T1)

        const wrapper2 = render(
            T1
        );
        expect(wrapper2).toMatchSnapshot();
        //component.find('#btnLog111').simulate('click');
        //expect(wrapper.find('#btnLog111')).toHaveLength(3);
        wrapper.find('#btnLog111').simulate('click');
        //console.log(1111,wrapper.find('div'))
        console.log(2222, toJson(wrapper))

        // component.find('img')
        // console.log(a)
        // console.log(2222222)

        // expect(a).toBe(5)
    })

    it.only('t1 demo', () => {
        let aa = <X1></X1>
        const component = mount(aa)
        // component.find('img')
        console.log(a)
        console.log(2222222)

        expect(a).toBe(5)
    })

})

// describe.each([
//     ['name', true],
//     [1, true],
//     [false, true],
//     [undefined, true],
//     ['no-such-key', false]
// ])('when key = %s', (key, expected) => {
//     it(`should return ${expected}`, () => {
//         expect(has(obj, key)).toBe(expected);
//     });
// });