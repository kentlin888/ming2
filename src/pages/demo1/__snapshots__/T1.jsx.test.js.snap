// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`T1.jsx.test.js t1 demo.....2 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "PPP2229",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "alt": "",
            "src": "../../assets/test-file-stub",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "attribs": Object {
                  "data-testid": "btnLog111",
                  "id": "btnLog111",
                },
                "children": Array [
                  Object {
                    "data": "log 111111",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "data-testid": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "data-testid": undefined,
                  "id": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "alt": undefined,
            "src": undefined,
          },
          "x-attribsPrefix": Object {
            "alt": undefined,
            "src": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {
          "alt": "",
          "src": "../../assets/test-file-stub",
        },
        "children": Array [],
        "name": "img",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "attribs": Object {
                "data-testid": "btnLog111",
                "id": "btnLog111",
              },
              "children": Array [
                Object {
                  "data": "log 111111",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "data-testid": undefined,
                "id": undefined,
              },
              "x-attribsPrefix": Object {
                "data-testid": undefined,
                "id": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "PPP2229",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "alt": undefined,
          "src": undefined,
        },
        "x-attribsPrefix": Object {
          "alt": undefined,
          "src": undefined,
        },
      },
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "data-testid": "btnLog111",
              "id": "btnLog111",
            },
            "children": Array [
              Object {
                "data": "log 111111",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "data-testid": undefined,
              "id": undefined,
            },
            "x-attribsPrefix": Object {
              "data-testid": undefined,
              "id": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "alt": "",
            "src": "../../assets/test-file-stub",
          },
          "children": Array [],
          "name": "img",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "PPP2229",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "alt": undefined,
            "src": undefined,
          },
          "x-attribsPrefix": Object {
            "alt": undefined,
            "src": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
